detectron2/data/detection_utils.py

```diff
@@ -73,7 +73,9 @@ def check_image_size(dataset_dict, image):
         expected_wh = (dataset_dict["width"], dataset_dict["height"])
         if not image_wh == expected_wh:
             raise SizeMismatchError(
-                "mismatch (W,H), got {}, expect {}".format(image_wh, expected_wh)
+                "mismatch (W,H), got {}, expect {} with name {}".format(
+                    image_wh, expected_wh, dataset_dict["file_name"]
+                )
```

Comment:
```
    name = dataset_dict.get("file_name")

    ....

                "Mismatched (W,H){}, got {}, expect {} with name {}".format(
                    " for image " + name if name is not None else "", image_wh, expected_wh
                )
```
so that it handles cases of no file_name.